src/test/java/uk/gov/register/core/RegisterContextTest.java

```diff
@@ -50,4 +54,40 @@ public void resetRegister_shouldResetRegister_whenEnableRegisterDataDeleteIsEnab
         verify(configManager, times(1)).refreshConfig();
         verify(flyway, times(1)).migrate();
     }
+
+    @Test
+    public void getRegisterMetadata_returnsUpToDateConfigProvidedByConfigManager() {
+        RegisterMetadata initialMetadata = new RegisterMetadata(
+                new RegisterName("test-register-1"),
+                Collections.emptyList(),
+                "copyright-1",
+                "registry-1",
+                "text-1",
+                "phase-1");
+
+        RegisterMetadata expectedUpdatedMetadata = new RegisterMetadata(
+                new RegisterName("test-register-2"),
+                Collections.emptyList(),
+                "copyright-2",
+                "registry-2",
+                "text-2",
+                "phase-2");
+
+
+        RegistersConfiguration rcMock = mock(RegistersConfiguration.class);
+
+        when(rcMock.getRegisterMetadata(registerName)).thenReturn(initialMetadata);
+        when(configManager.getRegistersConfiguration()).thenReturn(rcMock);
+
+        RegisterContext context = new RegisterContext(registerName, configManager, registerLinkService, dbi, flyway, Optional.empty(), true, false, Optional.empty(), Optional.empty(), emptyList(), new RegisterAuthenticator("", ""));
+
+        RegisterMetadata actualInitialMetadata = context.getRegisterMetadata();
+        assertThat(actualInitialMetadata, equalTo(initialMetadata));
+
+        when(rcMock.getRegisterMetadata(registerName)).thenReturn(expectedUpdatedMetadata);
+        when(configManager.getRegistersConfiguration()).thenReturn(rcMock);
```

Comment:
Instead of repeating these two lines, I think you could use `when(rcMock.getRegisterMetadata(registerName)).thenReturn(initialMetadata).thenReturn(expectedUpdatedMetadata)`. Removes a tiny bit of code :).